source/common/http/filter/ratelimit.cc

```diff
@@ -117,9 +117,16 @@ void Filter::complete(Envoy::RateLimit::LimitStatus status) {
     break;
   case Envoy::RateLimit::LimitStatus::OverLimit:
     cluster_->statsScope().counter("ratelimit.over_limit").inc();
-    Http::CodeUtility::ResponseStatInfo info{config_->scope(), cluster_->statsScope(), EMPTY_STRING,
-                                             *getTooManyRequestsHeader(), true, EMPTY_STRING,
-                                             EMPTY_STRING, EMPTY_STRING, EMPTY_STRING, false};
+    Http::CodeUtility::ResponseStatInfo info{config_->scope(),
```

Comment:
Yes I agree. Let me play with it a bit. There is something broken with header ordering also. I think I need to disable the header ordering feature. Let me play with it and I will tag you again.